pyproject.toml

```diff
@@ -99,17 +102,20 @@ commands =
     pytest: pytest --cov=src/pandablocks --cov-report term --cov-report xml:cov.xml {posargs}
     mypy: mypy src tests {posargs}
     pre-commit: pre-commit run --all-files {posargs}
+    type-checking: mypy src tests {posargs}
+    tests: pytest --cov=pandablocks --cov-report term --cov-report xml:cov.xml {posargs}
     docs: sphinx-{posargs:build -EW --keep-going} -T docs build/html
```

Comment:
A lot of errors when I use `tox -e docs`, I would guess there are still things in `docs/user` to be manually merged/deleted?

```
docs installed: accessible-pygments==0.0.4,alabaster==0.7.16,annotated-types==0.6.0,atomicwrites==1.4.1,Babel==2.14.0,beautifulsoup4==4.12.3,black==24.2.0,certifi==2024.2.2,cfgv==3.4.0,charset-normalizer==3.3.2,click==8.1.7,colorama==0.4.6,contourpy==1.2.0,copier==9.1.1,coverage==7.4.3,cycler==0.12.1,decorator==5.1.1,distlib==0.3.8,docutils==0.20.1,dunamai==1.19.2,filelock==3.13.1,fonttools==4.49.0,funcy==2.0,h5py==3.10.0,identify==2.5.35,idna==3.6,imagesize==1.4.1,iniconfig==2.0.0,Jinja2==3.1.3,jinja2-ansible-filters==1.3.2,kiwisolver==1.4.5,livereload==2.6.3,markdown-it-py==3.0.0,MarkupSafe==2.1.5,matplotlib==3.8.3,mdit-py-plugins==0.4.0,mdurl==0.1.2,mock==5.1.0,mypy==1.8.0,mypy-extensions==1.0.0,myst-parser==2.0.0,nodeenv==1.8.0,numpy==1.26.4,packaging==23.2,-e git+ssh://git@github.com/evalott100/PandABlocks-client.git@ae19b7187dcc244984327d75ac46ea6c75d2c059#egg=pandablocks,pathlib2==2.3.7.post1,pathspec==0.12.1,pillow==10.2.0,pipdeptree==2.15.1,platformdirs==4.2.0,pluggy==1.4.0,plumbum==1.8.2,pre-commit==3.6.2,prompt-toolkit==3.0.36,py==1.11.0,pydantic==2.6.3,pydantic_core==2.16.3,pydata-sphinx-theme==0.15.2,Pygments==2.17.2,pyparsing==3.1.1,pyright==1.1.351,pytest==8.0.2,pytest-asyncio==0.23.5,pytest-cov==4.1.0,python-dateutil==2.8.2,PyYAML==6.0.1,pyyaml-include==1.3.2,questionary==2.0.1,requests==2.31.0,ruff==0.2.2,six==1.16.0,snowballstemmer==2.2.0,soupsieve==2.5,Sphinx==7.2.6,sphinx-autobuild==2024.2.4,sphinx-copybutton==0.5.2,sphinx_design==0.5.0,sphinxcontrib-applehelp==1.0.8,sphinxcontrib-devhelp==1.0.6,sphinxcontrib-htmlhelp==2.0.5,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.7,sphinxcontrib-serializinghtml==1.1.10,tornado==6.4,tox==3.28.0,tox-direct==0.4,typed-ast==1.5.5,types-mock==5.1.0.20240106,typing_extensions==4.10.0,urllib3==2.2.1,virtualenv==20.25.1,wcwidth==0.2.13
docs run-test-pre: PYTHONHASHSEED='2227056538'
docs run-test: commands[0] | sphinx-build -EW --keep-going -T docs build/html
Running Sphinx v7.2.6
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://numpy.org/doc/stable/objects.inv...
loading intersphinx inventory from https://docs.h5py.org/en/stable/objects.inv...
myst v2.0.0: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions={'colon_fence'}, disable_syntax=[], all_links_external=False, url_schemes=('http', 'https', 'mailto', 'ftp'), ref_domains=None, fence_as_directive=set(), number_code_blocks=[], title_to_header=False, heading_anchors=0, heading_slug_func=None, html_meta={}, footnote_transition=True, words_per_minute=200, substitutions={}, linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', enable_checkboxes=False, suppress_warnings=[], highlight_code_blocks=True)
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 26 source files that are out of date
updating environment: [new config] 26 added, 0 changed, 0 removed
reading sources... [100%] user/tutorials/load-save
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/reference/contributing.rst:1: CRITICAL: Problems with "include" directive path:
InputError: [Errno 2] No such file or directory: '.github/CONTRIBUTING.rst'.
looking for now-outdated files... none found
pickling environment... done
checking consistency... /scratch/twj43146/Programming/PandABlocks-client-fork/docs/developer/explanations/decisions/0003-make-library-sans-io.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/introspect-panda.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/library-hdf.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/poll-changes.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/reference/changelog.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/reference/contributing.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/tutorials/commandline-hdf.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/tutorials/control.rst: WARNING: document isn't included in any toctree
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/tutorials/load-save.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
copying downloadable files... [100%] ../build/html/plot_directive/user/tutorials/commandline-hdf-2.pdf
copying static files... done
copying extra files... done
done
writing output... [100%] user/tutorials/load-save
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:53: WARNING: 'any' reference target not found: AsyncioClient.data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:53: WARNING: 'any' reference target not found: BlockingClient.data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:53: WARNING: 'any' reference target not found: StartData.fields
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:53: WARNING: 'any' reference target not found: write_hdf_files
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:70: WARNING: 'any' reference target not found: AsyncioClient.data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:70: WARNING: 'any' reference target not found: write_hdf_files
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:101: WARNING: 'any' reference target not found: DATA_OVERRUN
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:106: WARNING: 'any' reference target not found: DATA_OVERRUN
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:106: WARNING: 'any' reference target not found: FrameData
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:106: WARNING: 'any' reference target not found: AsyncioClient.data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:106: WARNING: 'any' reference target not found: BlockingClient.data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/performance.rst:113: WARNING: 'any' reference target not found: write_hdf_files
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:10: WARNING: 'any' reference target not found: BlockingClient
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:10: WARNING: 'any' reference target not found: AsyncioClient
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:18: WARNING: 'any' reference target not found: ControlConnection
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:18: WARNING: 'any' reference target not found: DataConnection
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:26: WARNING: py:meth reference target not found: ControlConnection.send
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:26: WARNING: 'any' reference target not found: Command
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:26: WARNING: py:meth reference target not found: ControlConnection.receive_bytes
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:26: WARNING: py:meth reference target not found: ControlConnection.responses
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:26: WARNING: 'any' reference target not found: Get
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:26: WARNING: 'any' reference target not found: GetFieldInfo
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:26: WARNING: 'any' reference target not found: FieldInfo
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:39: WARNING: py:meth reference target not found: DataConnection.connect
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:39: WARNING: py:meth reference target not found: DataConnection.receive_bytes
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:39: WARNING: 'any' reference target not found: Data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:39: WARNING: 'any' reference target not found: FrameData
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:39: WARNING: py:meth reference target not found: DataConnection.flush
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/explanations/sans-io.rst:54: WARNING: 'any' reference target not found: blocking
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/introspect-panda.rst:4: WARNING: 'any' reference target not found: pandablocks.commands
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/introspect-panda.rst:14: WARNING: 'any' reference target not found: BlockInfo
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/introspect-panda.rst:14: WARNING: 'any' reference target not found: GetBlockInfo
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/introspect-panda.rst:17: WARNING: 'any' reference target not found: FieldInfo
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/library-hdf.rst:6: WARNING: 'any' reference target not found: write_hdf_files
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/library-hdf.rst:18: WARNING: 'any' reference target not found: AsyncioClient
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/library-hdf.rst:18: WARNING: 'any' reference target not found: write_hdf_files
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/library-hdf.rst:33: WARNING: 'any' reference target not found: Data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/library-hdf.rst:41: WARNING: 'any' reference target not found: AsyncioClient.data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/library-hdf.rst:41: WARNING: 'any' reference target not found: Data
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/how-to/library-hdf.rst:41: WARNING: 'any' reference target not found: Pipeline
/scratch/twj43146/Programming/PandABlocks-client-fork/docs/user/reference/appendix.rst:11: WARNING: 'any' reference target not found: Command
generating indices... genindex py-modindex done
highlighting module code... [100%] pandablocks.connections
writing additional pages... search done
copying images... [100%] user/tutorials/tutorial_layout.png
dumping search index in English (code: en)... done
dumping object inventory... done
build finished with problems, 53 warnings.
ERROR: InvocationError for command /scratch/twj43146/Programming/PandABlocks-client-fork/.venv/bin/sphinx-build -EW --keep-going -T docs build/html (exited with code 1)
```